File: .sdk/tm/rb/utility/make_error.rb
# PassantenfrequenzenZuerich SDK utility: make_error
require_relative '../core/operation'
require_relative '../core/result'
require_relative '../core/error'
module PassantenfrequenzenZuerichUtilities
  MakeError = ->(ctx, err) {
    if ctx.nil?
      require_relative '../core/context'
      ctx = PassantenfrequenzenZuerichContext.new({}, nil)
    end
    op = ctx.op || PassantenfrequenzenZuerichOperation.new({})
    opname = op.name
    opname = "unknown operation" if opname.empty? || opname == "_"

    result = ctx.result || PassantenfrequenzenZuerichResult.new({})
    result.ok = false

    err = result.err if err.nil?
    err = ctx.make_error("unknown", "unknown error") if err.nil?

    errmsg = err.is_a?(PassantenfrequenzenZuerichError) ? err.msg : err.to_s
    msg = "PassantenfrequenzenZuerichSDK: #{opname}: #{errmsg}"
    msg = ctx.utility.clean.call(ctx, msg)

    result.err = nil
    spec = ctx.spec

    if ctx.ctrl.explain
      ctx.ctrl.explain["err"] = { "message" => msg }
    end

    sdk_err = PassantenfrequenzenZuerichError.new("", msg, ctx)
    sdk_err.result = ctx.utility.clean.call(ctx, result)
    sdk_err.spec = ctx.utility.clean.call(ctx, spec)
    sdk_err.code = err.code if err.is_a?(PassantenfrequenzenZuerichError)

    ctx.ctrl.err = sdk_err

    if ctx.ctrl.throw_err == false
      return result.resdata, nil
    end
    return nil, sdk_err
  }
end

File: py/utility/make_error.py
# PassantenfrequenzenZuerich SDK utility: make_error

from __future__ import annotations
from core.operation import PassantenfrequenzenZuerichOperation
from core.result import PassantenfrequenzenZuerichResult
from core.control import PassantenfrequenzenZuerichControl
from core.error import PassantenfrequenzenZuerichError


def make_error_util(ctx, err):
    if ctx is None:
        from core.context import PassantenfrequenzenZuerichContext
        ctx = PassantenfrequenzenZuerichContext({}, None)

    op = ctx.op
    if op is None:
        op = PassantenfrequenzenZuerichOperation({})
    opname = op.name
    if opname == "" or opname == "_":
        opname = "unknown operation"

    result = ctx.result
    if result is None:
        result = PassantenfrequenzenZuerichResult({})
    result.ok = False

    if err is None:
        err = result.err
    if err is None:
        err = ctx.make_error("unknown", "unknown error")

    errmsg = ""
    if isinstance(err, PassantenfrequenzenZuerichError):
        errmsg = err.msg
    elif hasattr(err, "msg") and err.msg is not None:
        errmsg = err.msg
    elif isinstance(err, str):
        errmsg = err
    else:
        errmsg = str(err)

    msg = "PassantenfrequenzenZuerichSDK: " + opname + ": " + errmsg
    msg = ctx.utility.clean(ctx, msg)

    result.err = None

    spec = ctx.spec

    if ctx.ctrl.explain is not None:
        ctx.ctrl.explain["err"] = {"message": msg}

    sdk_err = PassantenfrequenzenZuerichError("", msg, ctx)
    sdk_err.result = ctx.utility.clean(ctx, result)
    sdk_err.spec = ctx.utility.clean(ctx, spec)

    if isinstance(err, PassantenfrequenzenZuerichError):
        sdk_err.code = err.code

    ctx.ctrl.err = sdk_err

    if ctx.ctrl.throw_err is False:
        return result.resdata

    raise sdk_err
